evaluator/builtin_time.go

```diff
@@ -673,6 +675,326 @@ func builtinFromUnixTime(args []types.Datum, _ context.Context) (d types.Datum,
 	return builtinDateFormat([]types.Datum{d, args[1]}, nil)
 }
 
+// strToDate converts date string according to format, return true on success,
+// the value will be stored in argument t.
+func strToDate(t *time.Time, date string, format string) bool {
+	date = skipWhiteSpace(date)
+	format = skipWhiteSpace(format)
+
+	token, formatRemain, succ := getFormatToken(format)
+	if !succ {
+		return false
+	}
+
+	if token == "" {
+		return date == ""
+	}
+
+	dateRemain, succ := matchDateWithToken(t, date, token)
+	if !succ {
+		return false
+	}
+
+	return strToDate(t, dateRemain, formatRemain)
+}
+
+func getFormatToken(format string) (token string, remain string, succ bool) {
```

Comment:
why?
this is not an exported function and very easy to understand, just as the function name says~